src/auto/injector.js

```diff
@@ -337,7 +337,42 @@ function annotate(fn, strictDi, name) {
  *
  * @returns {Array.<string>} The names of the services which the function requires.
  */
-
+/**
+ * @ngdoc method
+ * @name $injector#loadNewModules
+ *
+ * @description
+ *
+ * **This is a dangerous API, which you use at your own risk!**
+ *
+ * Add the specified modules to the current injector.
+ *
+ * This method will add each of the injectables to the injector and execute all of the config and run
+ * blocks for each module passed to the method.
+ *
+ * If a module has already been loaded then it will not be loaded again.
+ *
+ * * The application developer is responsible for loading the code containing the modules.
+ * * Modules cannot be unloaded.
+ * * Previously compiled HTML will not be affected by newly loaded directives, filters and components.
+ *
+ * ## Example
+ *
+ * Here is an example of loading a bundle of modules, with a utility method called `getScript`:
+ *
+ * ```javascript
+ *  app.factory('loadModule', function($injector) {
+ *  return function loadModule(moduleName, bundleUrl) {
+ *    return getScript(bundleUrl).then(function() { $injector.loadNewModules([moduleName]); });
```

Comment:
You could use `fetch` in-place here for `getScript`. Only thing that might happen is that people use it and complain that it doesn't work in IE9 ...

Reply:
@Narretz I think the difference is that `getScript` also executes the script.